sdk/sdkservices/runtimes.go

```diff
@@ -48,6 +51,20 @@ type (
 	RunCallFunc  = func(ctx context.Context, rid sdktypes.RunID, v sdktypes.Value, args []sdktypes.Value, kwargs map[string]sdktypes.Value) (sdktypes.Value, error)
 	RunPrintFunc = func(ctx context.Context, rid sdktypes.RunID, text string)
 	NewRunIDFunc = func() sdktypes.RunID
+
+	RunSleepFunc       = func(ctx context.Context, t time.Duration) error
+	RunStartFunc       = func(ctx context.Context, entrypoint sdktypes.CodeLocation, inputs map[string]sdktypes.Value, memo map[string]string) (string, error)
+	RunSubscribeFunc   = func(ctx context.Context, name, filter string) (uuid.UUID, error)
+	RunUnsubscribeFunc = func(ctx context.Context, id uuid.UUID) error
+	RunNextEventFunc   = func(ctx context.Context, subscriptions []uuid.UUID, t time.Duration) (sdktypes.Value, error)
+
+	RunSyscalls struct {
```

Comment:
Why not use an interface. IMO even a concrete struct that holds the required state will be better/